al-nginx/nginx.conf

```diff
@@ -42,92 +42,38 @@ http {
             root /usr/share/nginx/html;
         }
 
-        location = /modelujeme/sluzby/auth-server {
-          return 302 /modelujeme/sluzby/auth-server/;
-        }
-
         location /modelujeme/sluzby/auth-server/ {
-            proxy_pass http://al-auth-server:8080;
-            proxy_set_header X-Forwarded-Host $host;
+            proxy_pass http://al-auth-server:8080; # no trailing slash to preserve matched prefix
```

Comment:
This is the biggest issue. Basically the whole path needs to be proxied to Keycloak so that it processess redirects accordingly. There is a way to tell Keycloak that it runs on a subpath, but a custom Dockerfile is neede to achieve that.